src/features/click-to-load.js

```diff
@@ -48,9 +52,18 @@ const readyToDisplayPlaceholders = new Promise(resolve => {
 let afterPageLoadResolver
 const afterPageLoad = new Promise(resolve => { afterPageLoadResolver = resolve })
 
-// Used to choose between extension/desktop flow or mobile apps flow.
-// Updated on ClickToLoad.init()
-let isMobileApp
+// Messaging layer for Click to Load
+/** @type {import("@duckduckgo/messaging").Messaging} */
+let _messagingModuleScope
+const lazy = {
```

Comment:
I'm not a huge fan of this, I figure it would be better to use the new messaging abstraction directly?

I've not looked at that, but I assume it would require making more things async, or at least thinking through more carefully about when the messaging system is ready for use. But I think that's better than pretending piercing the abstraction to pretend the messaging system is always ready.

Reply:
I agree but instead of making this code async we should just make is sync but only ever called after messaging is possible.

If that requires too much rewrite this is still acceptable to me so long as we never can get to the thrown error.

Reply:
The issue here is all the module-level state. Something to be avoided where possible going forward.

These problems arise when you have free-standing functions that actually need access to state (_state_ here meaning the instance of Messaging)

Not for this project, but we should eventually resolve with dependency injection, either:

a) wrapping all the stateful things in more classes, converting to methods instead of free-standing functions
b) or, just wrap everything in a big function and use the closure to access 'stateful' things that we make available via the functions params

When does either, we should also split this file to separate the framework initialisation (boundary) with the actual CTL functionality. That will help to clearly define what is stateful, and what isn't :)